File: ios/JJMproject/Classes/JSBridge/Managers/ProgressBar/ProgressBarModule.m
//
//  NetworkModule.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/14.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ProgressBarModule, NSObject)

RCT_EXTERN_METHOD(show)
RCT_EXTERN_METHOD(showWithMessage:(NSString *)message)
RCT_EXTERN_METHOD(dismiss)

@end

File: ios/JJMproject/Classes/JSBridge/Managers/ProgressBar/ProgressBarModule.swift
//
//  ProgressBarModule.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/11/14.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import UIKit
import Async

@objc(ProgressBarModule)
class ProgressBarModule: NSObject {
  
  private lazy var progressBarManager: ProgressBarManager = {
    return ProgressBarManager()
  }()
  
  @objc func show() {
    Async.main {
      self.progressBarManager.show()
    }
  }
  
  @objc func show(message: String) {
    Async.main {
      self.progressBarManager.show(message: message)
    }
  }
  
  @objc func dismiss() {
    Async.main {
      self.progressBarManager.dismiss()
    }
  }
  
}
